ludwig/data/dataset/parquet.py

```diff
@@ -0,0 +1,83 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2020 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import tensorflow as tf
+
+from petastorm import make_batch_reader
+from petastorm.tf_utils import make_petastorm_dataset
+
+from ludwig.constants import NAME, PROC_COLUMN
+from ludwig.data.batcher.iterable import IterableBatcher
+from ludwig.data.dataset.base import Dataset
+
+
+class ParquetDataset(Dataset):
+    def __init__(self, url, features, training_set_metadata):
+        self.url = url
+        self.training_set_metadata = training_set_metadata
+
+        with make_batch_reader(self.url) as reader:
+            self.size = reader.dataset.metadata.num_rows
+
+        self.reshape_features = {
+            feature[PROC_COLUMN]: list((-1, *training_set_metadata[feature[NAME]]['reshape']))
+            for feature in features
+            if 'reshape' in training_set_metadata[feature[NAME]]
+        }
+
+    def get(self, feature_name, sample):
+        t = getattr(sample, feature_name)
+        reshape_dim = self.reshape_features.get(feature_name)
+        if reshape_dim is not None:
+            t = tf.reshape(t, reshape_dim)
```

Comment:
I guess my previous question had to do with this, right? Tensors with multiple ranks get squashed and then put back in shape, right?

Reply:
Yes, exactly, I can add a comment here.